src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -219,6 +229,10 @@ public ClusterName getClusterName() {
         return this.clusterName;
     }
 
+    public boolean fromDiff() {
```

Comment:
I'm doubting a bit about this one. Should it be package private with a comment it's only used for testing? To the rest of the universe it shouldn't matter how the cluster state was constructed.